src/UserDataAccessLayer.cpp

```diff
@@ -0,0 +1,110 @@
+#include "UserData.hpp"
+#include "UserDataAccessLayer.hpp"
+#include <memory>
+#include <string>
+#include <cstdlib>
+#include "Group.hpp"
+#include <utility>
+
+namespace
+{
+
+std::optional<Group> inputToGroup(int input)
+{
+	switch(input)
+	{
+		case 1:
+			return std::optional<Group>{Group::weekend};
+		case 2:
+			return std::optional<Group>{Group::evening};
+		default:
+			std::cout << "Incorrect group";
+			std::cin.clear();
+			std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+			std::system("clear");
+			return std::nullopt;
+	}
```

Comment:
`default:
			std::cout << "Incorrect group";
			std::cin.clear();
			std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
			std::system("clear");
			return std::nullopt;`

I suggest add - sleep(time) - after std::cout in order to keep the notification on the screen for a bit of time.
Right now this notification is not even showing on the screen because it's instantly being erased.